src/storage/sstable.rs

```diff
@@ -198,24 +198,20 @@ impl SSTable {
     }
 }
 
-#[derive(Debug)]
-struct SparseIndex {
+#[derive(Deref, DerefMut, Debug)]
```

Comment:
what's the `Deref` on SparseIndex meant to enable?

Reply:
It allows `&OuterType` to be automatically convertible to `&InnerType`.

Reply:
I'm aware of the behavior of Deref in general. i'm curious why it's desired for the sparse index. (as this adt type is intended to be opaque. callers don't need to know it's internal implementation.)